src/scripts/contentCapture/augmentationHelper.ts

```diff
@@ -102,30 +103,34 @@ export class AugmentationHelper {
 	/*
 	 * Returns the augmented preview text.
 	 */
-	public static makeAugmentationRequest(url: string, locale: string, pageContent: string, requestCorrelationId: string): Promise<OneNoteApi.ResponsePackage<any>> {
-		return Clipper.getUserSessionIdWhenDefined().then((sessionId) => {
-			let augmentationApiUrl = Constants.Urls.augmentationApiUrl + "?renderMethod=extractAggressive&url=" + url + "&lang=" + locale;
-
-			let headers = {};
-			headers[Constants.HeaderValues.appIdKey] = Settings.getSetting("App_Id");
-			headers[Constants.HeaderValues.noAuthKey] = "true";
-			headers[Constants.HeaderValues.correlationId] = requestCorrelationId;
-			headers[Constants.HeaderValues.userSessionIdKey] = sessionId;
-
-			return HttpWithRetries.post(augmentationApiUrl, pageContent, headers).then((request: XMLHttpRequest) => {
-				let parsedResponse: any;
-				try {
-					parsedResponse = JSON.parse(request.response);
-				} catch (e) {
-					Clipper.logger.logJsonParseUnexpected(request.response);
-					return Promise.reject(OneNoteApi.ErrorUtils.createRequestErrorObject(request, OneNoteApi.RequestErrorType.UNABLE_TO_PARSE_RESPONSE));
-				}
-
-				let responsePackage = {
-					parsedResponse: parsedResponse,
-					request: request
-				};
-				return Promise.resolve(responsePackage);
+	public static makeAugmentationRequest(url: string, locale: string, pageContent: string, requestCorrelationId: string): Promise<ResponsePackage<any>> {
+		return new Promise<ResponsePackage<any>>((resolve, reject) => {
+			Clipper.getUserSessionIdWhenDefined().then((sessionId) => {
+				let augmentationApiUrl = Constants.Urls.augmentationApiUrl + "?renderMethod=extractAggressive&url=" + url + "&lang=" + locale;
+
+				let headers = {};
+				headers[Constants.HeaderValues.appIdKey] = Settings.getSetting("App_Id");
+				headers[Constants.HeaderValues.noAuthKey] = "true";
+				headers[Constants.HeaderValues.correlationId] = requestCorrelationId;
+				headers[Constants.HeaderValues.userSessionIdKey] = sessionId;
+
+				HttpWithRetries.post(augmentationApiUrl, pageContent, headers).then((response: Response) => {
+					response.text().then((responseText: string) => {
+						let parsedResponse: any;
+						try {
+							parsedResponse = JSON.parse(responseText);
+						} catch (e) {
+							Clipper.logger.logJsonParseUnexpected(responseText);
+							reject(OneNoteApi.RequestErrorType.UNABLE_TO_PARSE_RESPONSE);
```

Comment:
Will it be valuable while debugging failure conditions if we pass the response object in reject?
NOTE:- This applies to multiple files where reject has been used

Reply:
Sure. However, we cannot directly use `OneNoteApi.ErrorUtils.createRequestErrorObject` since this function accepts a variable of type `XMLHttpRequest` as its first argument. Hence, I have now created a copy of this function within the WebClipper repo itself such that it accepts a `Response` as its first argument, and have used this function instead.